apps/web/src/hooks/useRegisterNameCallback.ts

```diff
@@ -114,16 +157,35 @@ export function useRegisterNameCallback(
       ],
     });
 
+    let coinTypesForRequest: readonly bigint[] = [];
+    let signatureExpiryForRequest: bigint = 0n;
+    let signatureForRequest: `0x${string}` = '0x';
+
+    if (!paymasterServiceEnabled && reverseRecord) {
+      const payload =
+        reverseSigPayload ??
+        (await prepareReverseRecordSignature().catch((e) => {
+          logError(e, 'Reverse record signature step failed');
+          return null;
+        }));
+      if (!payload) return;
+      coinTypesForRequest = payload.coinTypes;
+      signatureExpiryForRequest = payload.signatureExpiry;
+      signatureForRequest = payload.signature;
+    }
+
+    const reverseRecordForRequest = paymasterServiceEnabled ? false : reverseRecord;
```

Comment:
Is this right? I think it should still be `reverseRecord` when `paymasterServiceEnabled` is true

Reply:
I think this is correct because when paymaster is enabled, we set reverseRecord through the SCW path batch call with a separate call entirely, so we need to pass false to the actual registrar contract to avoid the signature flow even if reverseRecord is true. we only want to pass true to the contract call if we're not on the SCW path AND we have `reverseRecord` == true

Reply:
Ah okay got it